hasjob/views/listing.py

```diff
@@ -174,23 +174,36 @@ def jobdetail(domain, hashid):
     else:
         g.starred_ids = set()
 
-    jobpost_ab = session_jobpost_ab()
-    related_posts = post.related_posts().all()
-    if is_siteadmin or (g.user and g.user.flags.get('is_employer_month')):
-        cache_viewcounts(related_posts)
     is_bgroup = getbool(request.args.get('b'))
     headline = post.headlineb if is_bgroup and post.headlineb else post.headline
-    g.impressions = {rp.id: (False, rp.id, bgroup(jobpost_ab, rp)) for rp in related_posts}
 
     return render_template('detail.html', post=post, headline=headline, reportform=reportform, rejectform=rejectform,
         pinnedform=pinnedform, applyform=applyform, job_application=job_application,
         jobview=jobview, report=report, moderateform=moderateform,
         domain_mismatch=domain_mismatch, header_campaign=header_campaign,
-        related_posts=related_posts, is_bgroup=is_bgroup,
-        is_siteadmin=is_siteadmin
+        is_bgroup=is_bgroup, is_siteadmin=is_siteadmin
         )
 
 
+@csrf.exempt
+@app.route('/<domain>/<hashid>/related', methods=('GET', 'POST'), subdomain='<subdomain>')
+@app.route('/<domain>/<hashid>/related', methods=('GET', 'POST'))
+@app.route('/view/<hashid>/related', defaults={'domain': None}, methods=('GET', 'POST'), subdomain='<subdomain>')
+@app.route('/view/<hashid>/related', defaults={'domain': None}, methods=('GET', 'POST'))
```

Comment:
In the original version of Hasjob back in 2011, the report form did an inline popup _without_ using JS, using a CSS `:target` selector to make the popup visible. Submitting the form would remove the target from the URL, so the popup would hide again.

While this is totally obsolete in the modern JS popup and form submission era, the old code has basically just stuck around.